lib/streamlit/util.py

```diff
@@ -204,6 +217,56 @@ def extract_key_query_params(
     }
 
 
+class BadTimeStringError(StreamlitAPIException):
+    """Raised when a bad time string argument is passed."""
+
+    def __init__(self, t: str):
+        MarkdownFormattedException.__init__(
+            self,
+            "Time string doesn't look right. It should be formatted as"
+            f"`'1d2h34m'` or `2 days`, for example. Got: {t}",
+        )
+
+
+@overload
+def time_to_seconds(
+    t: float | timedelta | str | None, *, coerce_none_to_inf: Literal[False]
+) -> float | None:
+    ...
+
+
+@overload
+def time_to_seconds(t: float | timedelta | str | None) -> float:
+    ...
+
+
+def time_to_seconds(
```

Comment:
nit: One consideration here could be to create a new `time_util` module. We have a couple of time-related utility methods in the codebase already that we could eventually move into this module. E.g. [adjust_years](https://github.com/streamlit/streamlit/blob/23f4ebf4125b3d63af5957345daef282195f0912/lib/streamlit/elements/widgets/time_widgets.py#L70). Maybe it is a good moment to create this module.

Reply:
Ah, yeah -- seems like a good time to do something like this. I created a new `time_util` module and moved `time_to_seconds` and `adjust_years` into it, but I'll hold off on moving any other time-related utilities into the module since it'll increase the size of the feature branch